### PR TITLE
Add support for AWS amd64/arm64 image build

### LINKED ISSUE #9: Add Support for building amd64 images on the AWS platform
### Describe the improvement request

The project should be able to support amd64 images on the AWS platform

### LINKED ISSUE #10: Add Support for building arm64 images on the AWS platform
### Describe the improvement request

The project should be able to support arm64 images on the AWS platform

### PR DESCRIPTION
**Pull request by obasekiosa**
This PR adds the packer scripts required to build, package and push an AWS ubuntu-focal amd64 image to AWS.

The images are kept up to date as the script uses the latest official release of the image version.

For a working set up the following environment variables need to be set 
```sh
AWS_ACCESS_KEY_ID=YOUR_ACCESS_KEY
AWS_SECRET_ACCESS_KEY=YOUR_SECRET_KEY
```

fixes #9 
fixes #10

### CODE REVIEW

File: aws/runner-machine-image.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+  tags = {
+      Name = "ghrunner"
+  }
+
+}
+
+
+build {
+  name = "ghrunner"
```
Comment (pavlovic-ivan) [inline aws/runner-machine-image.pkr.hcl]: @obasekiosa is the `name` necessary?

File: aws/runner-machine-image.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    filters = {
+      name                = var.source_image
+      root-device-type    = "ebs"
+      virtualization-type = "hvm"
+    }
+    most_recent = true
+    owners      = ["099720109477"]
+  }
+  ssh_username = "ubuntu"
+  run_tags = {
```
Comment (pavlovic-ivan) [inline aws/runner-machine-image.pkr.hcl]: You can remove `run_tags`

File: aws/runner-machine-image.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      Name = "ghrunner"
+  }
+  tags = {
```
Comment (pavlovic-ivan) [inline aws/runner-machine-image.pkr.hcl]: You can remove `tags` as well

**Comment by pavlovic-ivan** [comment]
@obasekiosa i would also suggest removing NVIDIA drivers variable, and also, create a new script, that will just download and install the runner. Then, you can reference that script in the packer file. The goal is to get the image just with the runner, and as light as possible, for starters, we can add drivers later

File: aws/runner-machine-image.pkr.hcl (hunk 1 above)
Comment (obasekiosa) [reply aws/runner-machine-image.pkr.hcl]: True it isn't, I'll remove it

File: aws/runner-machine-image.pkr.hcl (hunk 2 above)
Comment (obasekiosa) [reply aws/runner-machine-image.pkr.hcl]: If I do that the default tag on the tmp instance used to make the image would be "Packer Builder"

File: aws/runner-machine-image.pkr.hcl (hunk 3 above)
Comment (obasekiosa) [reply aws/runner-machine-image.pkr.hcl]: If I do that the default tag on the ami would be "Packer Builder".
Which would be a bit vague.

Is that ok or is there something I am missing?

**Comment by obasekiosa** [comment]
> @obasekiosa i would also suggest removing NVIDIA drivers variable, and also, create a new script, that will just download and install the runner. Then, you can reference that script in the packer file. The goal is to get the image just with the runner, and as light as possible, for starters, we can add drivers later

Ok, I will do this.
I'll also be releasing a PR for the AWS pulumi by today or Monday morning.

File: aws/runner-machine-image.pkr.hcl (hunk 3 above)
Comment (pavlovic-ivan) [reply aws/runner-machine-image.pkr.hcl]: ok, let me think about this

**Review by pavlovic-ivan — APPROVED** [review]
Looks good! Awesome work 🙏

**Review by pavlovic-ivan — APPROVED** [review]

**Pull request merged**